src/ol/renderer/canvas/canvasmaprenderer.js

```diff
@@ -193,11 +227,15 @@ ol.renderer.canvas.Map.prototype.renderFrame = function(frameState) {
       continue;
     }
     if (layerRenderer.prepareFrame(frameState, layerState)) {
+      if (layer instanceof ol.layer.Vector && layer.getSource().getWrapX()) {
```

Comment:
When editing vectors near the date line, you usually want a wrapped base layer, but your vector layer will have wrapX set to false so you can create geometries that cross the dateline.